[libc] Add simple features.h with implementation macro

alfredfo opened the pull request:
In the future this should probably be autogenerated so it defines library version.

Only added config/linux/* at the moment with an
if(LLVM_LIBC_FULL_BUILD) statement as I don't think this makes sense for overlay builds.

See: Discussion in #libc https://discord.com/channels/636084430946959380/636732994891284500/1163979080979460176

@michaelrj-google

michaelrj-google commented on libc/config/linux/riscv/headers.txt:
```diff
@@ -39,3 +39,9 @@ set(TARGET_PUBLIC_HEADERS
     libc.include.sys_wait
 )
+
+if(LLVM_LIBC_FULL_BUILD)
```
same here

michaelrj-google commented on libc/config/linux/x86_64/headers.txt:
```diff
@@ -39,3 +39,9 @@ set(TARGET_PUBLIC_HEADERS
     libc.include.sys_wait
 )
+
+if(LLVM_LIBC_FULL_BUILD)
```
same here

michaelrj-google commented on libc/include/llvm-libc-macros/features-macros.h:
```diff
@@ -0,0 +1,15 @@
+//===-- Definition of macros from features.h
+//----------------------------------===//
```
formatting

alfredfo replied:
github CI got angry at me for not doing this before :) Will try again

michaelrj-google commented on libc/include/features.h.def:
```diff
@@ -0,0 +1,17 @@
+//===-- C standard library header features.h ---------------------------------===//
```
formatting

michaelrj-google reviewed (commented):
The patch is good with some nits

alfredfo commented on libc/config/linux/aarch64/headers.txt:
```diff
@@ -19,3 +19,9 @@ set(TARGET_PUBLIC_HEADERS
     libc.include.unistd
 )
+
+if(LLVM_LIBC_FULL_BUILD)
```
Right, will remove

jhuber6 commented on libc/include/llvm-libc-macros/features-macros.h:
```diff
@@ -0,0 +1,15 @@
+#ifndef __LLVM_LIBC_MACROS_FEATURES_MACROS_H
+#define __LLVM_LIBC_MACROS_FEATURES_MACROS_H
+
+#define __LLVMLIBC__ 1
```
Why `__LLVMLIBC__` instead of `__LLVM_LIBC__`? Is there some prior art here?

alfredfo replied:
I just picked one, also wondered whether `LLVM_LIBC` or `LLVMLIBC` fit better here. Could change it to __LLVM_LIBC__ if you prefer

alfredfo replied:
Sounds good, merged with `__LLVM_LIBC__`

alfredfo committed:
[libc] Add simple features.h with implementation macro

In the future this should probably be autogenerated so it defines library version.

Only added config/linux/* at the moment with an
if(LLVM_LIBC_FULL_BUILD statement as I don't think this makes sense for overlay builds.

See: Discussion in #libc https://discord.com/channels/636084430946959380/636732994891284500/1163979080979460176

michaelrj-google reviewed (approved):
overall LGTM

alfredfo commented on libc/include/features.h.def:
```diff
@@ -0,0 +1,17 @@
+//===-- C standard library header features.h -------------------------------===//
```
fixed: https://github.com/llvm/llvm-project/commit/67770cbb9841e86357021e03753a1b8f4ce783ea
